### ISSUE TITLE
extractor trick leaves geyser inaccessible

### ISSUE THREAD

**Issue by jjs47** [bwapi/bwapi#697]
BWAPI 4.1.2. When I iterate through getStaticGeysers(), only accessible geysers (with exists() and isVisible() true) return getType() == Resource_Vespene_Geyser. Others return getType() == 233--I'm not sure what that is supposed to mean, but it seems more or less as documented. But when my bot pulls the extractor trick on a geyser, then after canceling the extractor, iterating through getStaticGeysers() says that the same geyser has getType() == 233, exists() == false, and isVisible()==false; it is treated exactly as if it were inaccessible, even though it's still in plain sight.

**Comment by heinermann** [comment]
This happens for all races who cancel their refinery structure.

**Comment by heinermann** [comment]
The *actual* geyser appears in `getAllUnits()`, and the one in `getStaticGeysers` becomes a dummy. Not sure where it's duplicating the units though. Needs further investigation.

**Comment by jjs47** [comment]
I have seen the same thing happen when the extractor is destroyed in combat.

**Comment by heinermann** [comment]
Yes it is fundamentally the same issue. Possibly to do with the on destroy hook.

**Comment by N00byEdge** [comment]
Closing this as this is just how BW does things. Kinda sucks, yeah.

**Comment by heinermann** [comment]
I'll leave this one open since it's a legitimate issue, we just need to find a solution to deal with it correctly.
